cpp/src/parquet/arrow/writer.cc

```diff
@@ -1079,6 +1080,7 @@ class FileWriter::Impl {
     int current_column_idx = row_group_writer_->current_column();
     std::shared_ptr<::arrow::Schema> arrow_schema;
     RETURN_NOT_OK(FromParquetSchema(writer_->schema(), {current_column_idx - 1},
```

Comment:
Could this be avoided by using the `arrow::Schema` property on the `FileWriter`?  I read this as we are recomputing the mapping from parquet to arrow when we already have the original schema passed in to create the writer instance.  Eliminating this would resolve the unexpected coupling between the writer code on the reader.